electron/src/ipc-communication.ts

```diff
@@ -29,26 +34,88 @@ class IpcCommunication implements IpcRegister {
             throw new Error('Window was null!');
         }
 
-        ipcMain.on('get-images', (ev, par: string[]) => {
-            const dir = par[0];
+        ipcMain.on('get-images', (ev, dir: string) => {
             if (!dir) {
                 window.webContents.send('images-found', []);
                 return;
             }
-            fs.readdir(dir, (err, files) => {
+            fs.readdir(dir, (err, fileNames) => {
                 if (!!err) {
                     console.error(err);
                 }
-                files = files
+                const imgsData = fileNames
                     .map((name) => path.join(dir, name))
-                    .filter((imgDir) => {
-                        const imgChunk = readChunk.sync(imgDir, 0, 12);
-                        return imageType(imgChunk)?.mime.startsWith('image') ?? false;
-                    });
-                window.webContents.send('images-found', files);
+                    .filter((name) => this.filterFilesForImages(name))
+                    .map((name) => this.mapToImageData(name));
+                window.webContents.send('images-found', imgsData);
             });
         });
     }
+
+    private getImagesNumber(window: BrowserWindow | null): void {
+        if (!window) {
```

Comment:
We should extract this check to _registerIpcHandlers(window: Electron.BrowserWindow)_ and we will not need to check this condition.